Common/src/main/java/at/petrak/hexcasting/common/blocks/circles/BlockEntitySlate.java

```diff
@@ -42,4 +48,39 @@ protected void loadModData(CompoundTag tag) {
         }
     }
 
+    @Override
+    public CompoundTag readIotaTag(){
+        if(pattern == null){
+            return IotaType.serialize(emptyIota());
+        }
+        return IotaType.serialize(new PatternIota(pattern));
+    }
+
+    @Override
+    public Iota readIota(ServerLevel world) {
+        if(pattern == null){
+            return emptyIota();
+        }
+        return new PatternIota(pattern);
+    }
+
+    @Override
+    public boolean writeIota(@Nullable Iota iota, boolean simulate){
+        if(!simulate){
+            if(iota instanceof PatternIota pIota){
+                this.pattern = pIota.getPattern();
+                sync();
+            }
+            if(iota instanceof NullIota || iota == null){
+                this.pattern = null;
+                sync();
+            }
+        }
+        return iota instanceof PatternIota || iota instanceof NullIota || iota == null;
```

Comment:
This feels weirdly duplicated, but idk how to fix it (~~other than moving to Kotlin~~).